src/main/java/spoon/reflect/visitor/CtBiScannerDefault.java

```diff
@@ -28,7 +36,35 @@
  * Is used by EqualsVisitor.
  */
 // autogenerated by CtBiScannerGenerator
-public abstract class CtBiScannerDefault extends spoon.reflect.visitor.CtAbstractBiScanner {
+public class CtBiScannerDefault extends spoon.reflect.visitor.CtAbstractBiScanner {
+	protected Deque<CtElement> stack = new ArrayDeque<>();
+
+	protected void enter(CtElement e) {
+	}
+
+	protected void exit(CtElement e) {
+	}
+
+	public void biScan(CtElement element, CtElement other) {
+		stack.push(other);
+		try {
+			element.accept(this);
+		} finally {
+			stack.pop();
+		}
+	}
+
+	public void biScan(CtRole role, CtElement element, CtElement other) {
+		biScan(element, other);
+	}
+
+	protected boolean biScan(CtRole role, Collection<? extends CtElement> elements, Collection<? extends CtElement> others) {
+		for (Iterator<? extends CtElement> firstIt = elements.iterator(), secondIt = others.iterator(); (firstIt.hasNext()) && (secondIt.hasNext());) {
```

Comment:
I guess this code does not work for `Set` since beginning, because you cannot assure that elements in 2 Sets are iterated in consistent pairs. I wonder why it works ... WDYT?